### PR TITLE
Adding support for Github Enterprise, refactoring to use ES5 code and adding .gitignore

### PR DESCRIPTION
**Pull request by ankurp**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -3,13 +3,13 @@ var parseGitUrl = require('git-url-parse');
 
 module.exports.exec = function exec(command) {
   console.log("   executing: " + command);
-  const options = { silent: true };
-  const ref = shell.exec(command, options);
+  var options = { silent: true };
```
Comment (ankurp) [inline src/utils.js]: Being consistent by using `var` instead of `const` as this file has `ES5` syntax.

File: src/utils.js [hunk 2]
```diff
@@ -18,6 +18,12 @@ module.exports.exec = function exec(command) {
 
 module.exports.getGHPagesUrl = function getGHPagesUrl(ghUrl) {
   var parsedUrl = parseGitUrl(ghUrl);
-  var ghPagesUrl = 'https://' + parsedUrl.owner + '.github.io/' + parsedUrl.name + '/';
+  var ghPagesUrl;
+  if (parsedUrl.resource === 'github.com') {
```
Comment (ankurp) [inline src/utils.js]: Adding support for github enterprise where the URL domain does not contain `github.io`.

File: src/utils.js [hunk 3]
```diff
@@ -18,6 +18,12 @@ module.exports.exec = function exec(command) {
...
+    ghPagesUrl = 'https://' + parsedUrl.owner + '.github.io/' + parsedUrl.name + '/';
+  } else { // Github Enterprise
+    ghPagesUrl = 'https://' + parsedUrl.resource + '/pages/' + parsedUrl.full_name + '/';
```
Comment (ankurp) [inline src/utils.js]: Format for github enterprise pages is as such: `https://github-entperise-domain.com/pages/project-owner/repo-name/`

**Comment by ankurp** [comment]
Good to merge?

**Comment by ankurp** [comment]
Resolved merge conflict. Good to review again.

**Comment by ankurp** [comment]
I have rebase and resolved conflicts.

**Comment by orta** [comment]
Seriously though, changes look 👍

**Pull request merged**